crates/turbopack-core/src/resolve/mod.rs

```diff
@@ -95,61 +97,68 @@ impl ModuleResolveResult {
         affecting_sources: Vec<Vc<Box<dyn Source>>>,
     ) -> ModuleResolveResult {
         ModuleResolveResult {
-            primary: Vec::new(),
+            primary: IndexMap::new(),
             affecting_sources,
         }
     }
 
-    pub fn module(module: Vc<Box<dyn Module>>) -> ModuleResolveResult {
+    pub fn module(request_key: RequestKey, module: Vc<Box<dyn Module>>) -> ModuleResolveResult {
```

Comment:
Given this (and maybe others here or in ResolveResult as well) are mostly called with `RequestKey::default()`, maybe we should default to that here and add a new `module_with_key` function?

```suggestion
    pub fn module(module: Vc<Box<dyn Module>>) -> ModuleResolveResult {
        Self::module_with_key(RequestKey::default(), module)
    }

    pub fn module_with_key(request_key: RequestKey, module: Vc<Box<dyn Module>>) -> ModuleResolveResult {
```

Reply:
Would also make the diff a bit smaller 🙃